common/persistence/dataInterfaces.go

```diff
@@ -1196,6 +1223,26 @@ func UnixMilliseconds(t time.Time) int64 {
 	return unixNano / int64(time.Millisecond)
 }
 
+func ParseHistoryBranchToken(branchToken []byte) (*persistencespb.HistoryBranch, error) {
+	return serialization.HistoryBranchFromBlob(branchToken, enumspb.ENCODING_TYPE_PROTO3.String())
+}
+
+func UpdateHistoryBranchToken(branchToken []byte, branchInfo *persistencespb.HistoryBranch) ([]byte, error) {
+	bi, err := serialization.HistoryBranchFromBlob(branchToken, enumspb.ENCODING_TYPE_PROTO3.String())
```

Comment:
For OSS impl, this step is unnecessary as all three defined fields are overwritten below?

Reply:
I'd argue this is much safer from a future-proof stand point. Branch tokens "may" evolve to include more stuff in the future. The intent here is to only overwrite the branch info portion of the branch token. Therefore we shouldn't assume branch info === branch token (logically speaking)

Reply:
So the idea is that branchToken may contain some persistence store-specific information which may not in history branch definition? Even if that store-specific info is defined in OSS?